prometheus-to-sd/main.go

```diff
@@ -103,48 +98,26 @@ func main() {
 	<-make(chan int)
 }
 
-func extractSourceConfigsFromFlags() []config.SourceConfig {
-	var sourceConfigs []config.SourceConfig
-	for _, c := range source {
-		if sourceConfig, err := config.ParseSourceConfig(c); err != nil {
-			glog.Fatalf("Error while parsing source config flag %v: %v", c, err)
-		} else {
-			sourceConfigs = append(sourceConfigs, *sourceConfig)
-		}
-	}
-
-	if len(source) == 0 && *component != "" {
-		glog.Warningf("--component, --host, --port and --whitelisted flags are deprecated. Please use --source instead.")
-		portStr := strconv.FormatUint(uint64(*port), 10)
-
-		if sourceConfig, err := config.NewSourceConfig(*component, *host, portStr, *whitelisted); err != nil {
-			glog.Fatalf("Error while parsing --component flag: %v", err)
-		} else {
-			glog.Infof("Created a new source instance from --component flag: %+v", sourceConfig)
-			sourceConfigs = append(sourceConfigs, *sourceConfig)
-		}
-	}
-	return sourceConfigs
-}
-
 func readAndPushDataToStackdriver(stackdriverService *v3.Service, gceConf *config.GceConfig, podConfig *config.PodConfig, sourceConfig config.SourceConfig) {
 	glog.Infof("Running prometheus-to-sd, monitored target is %s %v:%v", sourceConfig.Component, sourceConfig.Host, sourceConfig.Port)
-
+	commonConfig := &config.CommonConfig{
+		GceConfig:     gceConf,
+		PodConfig:     podConfig,
+		ComponentName: sourceConfig.Component,
+	}
+	metricDescriptorCache := translator.NewMetricDescriptorCache(stackdriverService, commonConfig, sourceConfig.Component)
 	signal := time.After(0)
 	useWhitelistedMetricsAutodiscovery := *autoWhitelistMetrics && len(sourceConfig.Whitelisted) == 0
 
 	for range time.Tick(*resolution) {
+		metricDescriptorCache.MarkStale()
```

Comment:
They are not stale only if cache was refreshed during current iteration. I have added a comment to make it a bit more clear.